cmd/sysl/cmd_mod.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const syslModuleName = "syslmodules"
+
+type modCmd struct {
+	subcommand string
+}
+
+func (m *modCmd) Name() string       { return "mod" }
+func (m *modCmd) MaxSyslModule() int { return 0 }
+
+func (m *modCmd) Configure(app *kingpin.Application) *kingpin.CmdClause {
+	opts := []string{"init"}
+	cmd := app.Command(m.Name(), "Configure sysl module system")
+	cmd.Arg("command", fmt.Sprintf("commands: [%s]", strings.Join(opts, ","))).Required().EnumVar(&m.subcommand, opts...)
+
+	return cmd
+}
+
+func (m *modCmd) Execute(args ExecuteArgs) error {
+	if m.subcommand == "init" {
+		err := syslModInit(args)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
use `switch`
```go
switch m.subcommand {
case "init":
  return syslModInit(args)
default:
  return errors.New("command not recognized") // don't think this should be nil since no command was executed
}
```

Reply:
Pretty sure the kingpin cli parser filters it out, so it shouldnt be possible for subcommand to be anything than the required enum arguments. Ill change it anyway.